pkg/maps/metricsmap/metricsmap.go

```diff
@@ -86,6 +169,72 @@ var (
 		})
 )
 
+// updatePrometheusMetrics checks every key value pair
+// and determines which promethues metrics along with respective labels
+// need to be updated.
+func updatePrometheusMetrics(key *Key, val *Value) {
+	if key.IsDrop() {
```

Comment:
The `DropCount ` and `ForwardCount` take different labels as arguments.

How about this:
```
// UpdatePrometheusMetrics checks the metricsmap key value pair
// and determines which prometheus metrics along with respective labels
// need to be updated.
func UpdatePrometheusMetrics(key *metricsmap.Key, val *metricsmap.Value) {
	var counter prometheus.Counter
	var err error = nil
	if key.IsDrop() {
		counter, err = DropCount.GetMetricWithLabelValues(key.DropForwardReason(), key.Direction())
	} else {
		counter, err = ForwardCount.GetMetricWithLabelValues(key.Direction())
	}
	if err != nil {
		log.Warnf("failed to update prometheus metrics err:",err)
		return
	}
	oldValue := getCounterValue(counter)
	newValue := val.CountFloat()
	// Check if metrics have changed since the last poll.
	// If yes, we need to add only the delta.
	if newValue > oldValue {
		if key.IsDrop() {
			DropCount.WithLabelValues(key.DropForwardReason(), key.Direction()).Add((newValue - oldValue))
		} else {
			ForwardCount.WithLabelValues(key.Direction()).Add((newValue - oldValue))
		}
	}
}

```

Reply:
Looks good!

Reply:
done